File: urbanairship-core/src/main/java/com/urbanairship/modules/location/LocationModuleFactory.java
/* Copyright Airship and Contributors */

package com.urbanairship.modules.location;

import android.content.Context;

import com.urbanairship.AirshipVersionInfo;
import com.urbanairship.PreferenceDataStore;
import com.urbanairship.PrivacyManager;
import com.urbanairship.analytics.Analytics;
import com.urbanairship.channel.AirshipChannel;

import androidx.annotation.NonNull;
import androidx.annotation.RestrictTo;

/**
 * Location module factory.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public interface LocationModuleFactory extends AirshipVersionInfo {

    @NonNull
    LocationModule build(@NonNull Context context,
                         @NonNull PreferenceDataStore dataStore,
                         @NonNull PrivacyManager privacyManager,
                         @NonNull AirshipChannel airshipChannel,
                         @NonNull Analytics analytics);

}

File: urbanairship-core/src/main/java/com/urbanairship/modules/location/LocationModuleFactory.kt
/* Copyright Airship and Contributors */
package com.urbanairship.modules.location

import android.content.Context
import androidx.annotation.RestrictTo
import com.urbanairship.AirshipVersionInfo
import com.urbanairship.PreferenceDataStore
import com.urbanairship.PrivacyManager
import com.urbanairship.channel.AirshipChannel
import com.urbanairship.permission.PermissionsManager

/**
 * Location module factory.
 *
 * @hide
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
public interface LocationModuleFactory : AirshipVersionInfo {

    public fun build(
        context: Context,
        dataStore: PreferenceDataStore,
        privacyManager: PrivacyManager,
        airshipChannel: AirshipChannel,
        permissionsManager: PermissionsManager
    ): LocationModule
}
